modules/oeFax/controller/RCFaxClient.php

```diff
@@ -0,0 +1,624 @@
+            $timePerCallLogRequest = 10;
+            $flag = true;
+            // dateFrom and dateTo paramteres
+            $timeFrom = '00:00:00.000Z';
+            $timeTo = '23:59:59.000Z';
+            // Array to push the call-logs to a file
+            $callLogs = array();
+            $responseMsgs = "";
+
+            while ($flag) {
+                // Start Time
+                $start = microtime(true);
+                $dateFrom = $fromDate . 'T' . $timeFrom;
+                $dateTo = $toDate . 'T' . $timeTo;
+
+                $apiResponse = $this->platform->get('/account/~/extension/~/call-log', array(
+                    'dateFrom' => $dateFrom,
+                    'dateTo' => $dateTo,
+                    //'type' => 'SMS',
+                    'perPage' => 500,
+                    'page' => $pageCount
+                ));
+
+                $apiResponseArray = $apiResponse->json()->records;
+                $responseMsgs = '';
+                foreach ($apiResponseArray as $value) {
+                    $responseMsgs .= "<tr><td>" . str_replace(array("T", "Z"), " ", $value->startTime) . "</td><td>" . $value->type . "</td><td>" . $value->from->name . "</td><td>" . $value->to->name . "</td><td>" . $value->action . "</td><td>" . $value->result . "</td><td>" . $value->message->id . "</td></tr>";
+                    array_push($callLogs, $value);
+                }
+                $end = microtime(true);
+                // Check if the recording completed wihtin 10 seconds.
+                $time = ($end * 1000 - $start * 1000) / 1000;
+
+                // Check if 'nextPage' exists
+                if (isset($apiResponseJSONArray["navigation"]["nextPage"])) {
+                    if ($time < $timePerCallLogRequest) {
+                        sleep($timePerCallLogRequest - $time);
+                        sleep(5);
+                        $pageCount = $pageCount + 1;
+                    }
+                } else {
+                    $flag = false;
+                    unset($callLogs);
+                }
+            }
+        } catch (ApiException $e) {
+            $message = $e->getMessage();
+            return xlt('HTTP Error') . ': ' . $message . PHP_EOL;
+        }
+
+        return $responseMsgs;
+    }
+
+    public function getPending()
+    {
+        $dateFrom = $this->getRequest('datefrom');
+        $dateTo = $this->getRequest('dateto');
+
+        if (!$this->authenticate()) {
+            $e = xlt('Error: Authentication Service Denies Access.');
+            $ee = array('error' => $e);
+            return json_encode($ee);
+        };
+        try {
+            // Writing the call-log response to json file
+            $dir = 'fax';
+            $messageStoreDir = $this->baseDir; // . DIRECTORY_SEPARATOR . $dir;
+
+            //Create the Directory
+            if (!file_exists($messageStoreDir)) {
+                mkdir($messageStoreDir, 0777, true);
+            }
```

Comment:
Note only directories/subdirectories within the documents folder are writable (and secure from web prying eyes)